kazoo/testing/harness.py

```diff
@@ -208,8 +212,29 @@ def setup_zookeeper(self, **client_options):
             self.cluster.start()
         namespace = "/kazootests" + uuid.uuid4().hex
         self.hosts = self.servers + namespace
-        if "timeout" not in client_options:
-            client_options["timeout"] = self.DEFAULT_CLIENT_TIMEOUT
+
+        tries = 0
+        while True:
+            try:
+                client_cluster_health = self._get_client()
+                client_cluster_health.start()
+                client_cluster_health.ensure_path("/")
+                client_cluster_health.stop()
+                self.log(logging.INFO, "cluster looks ready to go")
+                break
+            except Exception:
+                tries += 1
+                if tries >= MAX_INIT_TRIES:
+                    raise
+                if tries > 0 and tries % 2 == 0:
+                    self.log(
+                        logging.WARNING,
+                        "nuking current cluster and making another one",
+                    )
+                    self.cluster.terminate()
+                    self.cluster.start()
```

Comment:
Is this nuking the underlying ZK cluster, or just the kazoo client trying to connect to the cluster? If the former, that's fairly expensive/slow, so do we need to sleep/pause a bit for it to get going before continuing? Or is that already handled internally within `cluster.start()`? (probably a better place for it)

Reply:
Yes, there is already some `sleep()` hidden in `cluster.start()` (https://github.com/python-zk/kazoo/blob/master/kazoo/testing/common.py#L370)